AsyncNats/NatsConnection.cs

```diff
@@ -278,9 +324,12 @@ private async Task ProcessMessagesAsync(PipeReader reader, CancellationToken dis
                                 break;
 
                             case NatsInformation info:
-                                _logger?.LogTrace("Received connection information for {Server}, {ConnectionInformation}", Options.Server, info);
+                                _logger?.LogTrace("Received connection information for {Server}, {ConnectionInformation}", _currentServer, info);
 
                                 NatsInformation = info;
+
+                                _serverPool.AddDiscoveredServers(info.ConnectURLs);
```

Comment:
Should there be an option to re-connect after discovery?

Reply:
I honestly don't see the benefit. Do you think there might be use cases that require it? It would mean breaking an existing working connection.  I would until someone comes with a valid use case to implement such logic.
The main intention of server discovery is to allow clients to adapt to topology changes. Let's say a nats cluster get its servers count expanded, clients reconnecting can now connect to the new servers and keep the cluster balanced, for example.

Reply:
Only reason I could think of to add this is that developers are lazy and if there's a cluster, they might only add 1 server.

Reply:
If there’s a cluster and you managed to successfully connect to a node, there’s no need to drop that connection. Unless geo topology discovery were a thing, but nats does not have that concept exposed in their clusters.